dask/dataframe/reshape.py

```diff
@@ -59,3 +60,76 @@ def get_dummies(data, prefix=None, prefix_sep='_', dummy_na=False,
                           prefix_sep=prefix_sep, dummy_na=dummy_na,
                           columns=columns, sparse=sparse,
                           drop_first=drop_first)
+
+
+def pivot_table(df, index=None, columns=None,
+                values=None, aggfunc='mean'):
+    """
+    Create a spreadsheet-style pivot table as a DataFrame. Target ``columns``
+    must have category dtype to infer result's ``columns``.
+    ``index``, ``columns``, ``values`` and ``aggfunc`` must be all scalar.
+
+    Parameters
+    ----------
+    data : DataFrame
+    values : scalar
+        column to aggregate
+    index : scalar
+        column to be index
+    columns : scalar
+        column to be columns
+    aggfunc : {'mean', 'sum', 'count'}, default 'mean'
+
+    Returns
+    -------
+    table : DataFrame
+    """
+
+    if not is_scalar(index) or index is None:
+        raise ValueError("'index' must be a scalar")
+    if not is_scalar(columns) or columns is None:
+        raise ValueError("'columns' must be a scalar")
+    if not is_categorical_dtype(df[columns]):
+        raise ValueError("'columns' must be category dtype")
+    if not is_scalar(values) or values is None:
+        raise ValueError("'values' must be a scalar")
+    if not is_scalar(aggfunc):
+        raise ValueError("'aggfunc' must be a scalar")
+    if aggfunc not in ('mean', 'sum', 'count'):
+        raise ValueError("aggfunc muset be either 'mean', 'sum', 'count'")
+
+    def agg(df):
+        return df.groupby(level=0).sum()
+
+    def pivot_sum(df, index=index, columns=columns,
+                  values=values):
+        return pd.pivot_table(df, index=index, columns=columns,
+                              values=values, aggfunc='sum')
+
+    def pivot_count(df, index=index, columns=columns,
+                    values=values):
+        return pd.pivot_table(df, index=index, columns=columns,
+                              values=values, aggfunc='count')
```

Comment:
It would be nice to move these functions to the top-level.  Dynamically generated functions are more expensive to serialize.  This ends up being important in the distributed computing case.